pkg/cloud/scope/machine.go

```diff
@@ -63,25 +63,28 @@ func NewMachineScope(params MachineScopeParams) (*MachineScope, error) {
 		params.Logger = klogr.New()
 	}
 
+	helper, err := patch.NewHelper(params.AWSMachine, params.Client)
+	if err != nil {
+		return nil, errors.Errorf("failed to init patch helper: %v", err)
+	}
 	return &MachineScope{
-		client:     params.Client,
-		patch:      client.MergeFrom(params.AWSMachine.DeepCopy()),
-		Cluster:    params.Cluster,
-		Machine:    params.Machine,
-		AWSMachine: params.AWSMachine,
-		Logger:     params.Logger,
+		client:      params.Client,
+		Cluster:     params.Cluster,
+		Machine:     params.Machine,
+		AWSMachine:  params.AWSMachine,
+		Logger:      params.Logger,
+		patchHelper: helper,
 	}, nil
 }
 
 // MachineScope defines a scope defined around a machine and its cluster.
 type MachineScope struct {
 	logr.Logger
-	patch  client.Patch
```

Comment:
A lot of times we separate the unexported from the exported fields, as was the case here. I'd recommend you keep this pattern - move `patchHelper` back up here, and preserve the blank line in between `client` and `Cluster`.

Reply:
ok, i'll keep that pattern